com.unity.ml-agents/Runtime/Policies/BehaviorParameters.cs

```diff
@@ -54,41 +54,41 @@ public BrainParameters brainParameters
 
         /// <summary>
         /// The neural network model used when in inference mode.
-        /// This cannot be set directly; use <see cref="Agent.GiveModel(string,NNModel,InferenceDevice)"/>
-        /// to set it.
+        /// This cannot be set at runtime; use <see cref="Agent.SetModel(string,NNModel,InferenceDevice)"/>
+        /// to set it instead.
         /// </summary>
         public NNModel model
         {
             get { return m_Model; }
-            internal set { m_Model = value; }
+            set { AccessUtilities.SetPropertyIfAllowed(ref m_Model, value); }
```

Comment:
Because of this change, I think it is now impossible to do Agent.SetModel at runtime since we use this setter : 
https://github.com/Unity-Technologies/ml-agents/pull/3602/files#diff-465b71e5c8718d9250bf7ba6abdea20cR371
One solution would be to have an internal setter for the model and behaviorType...

Reply:
Should be irrelevant now. 

Changing any of (behavior name, model, inference device) will now update the Agent's policy. So if you're going to change more than 1, it's slightly more efficient to use Agent.SetModel().